### PR TITLE
Fixed OOM Killed memory issue

### PR DESCRIPTION
**Pull request by redhatHameed**
This PR addresses a few changes needed for OOM issues identified when installing to environments with very heavy production load:

1- Resource limits are increased slightly
2- labels are attached to secrets so that they can be appropriately targeted via label selectors for manager's cache
3- manager's cache is configured to use secret filtering in order to prevent caching all secrets on cluster

After restricting the controller manager cache to the secrets, the memory usage dropped significantly see the graph with the yellow line. the blue line shows the memory usage before this change.
 

Screen Shot 2022-03-28 at 11 56 53 AM

local test run successfully

```
ahameed@ahameed-mac crunchy-bridge-operator % make test
/Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/bin/controller-gen "crd:trivialVersions=true,preserveUnknownFields=false" rbac:roleName=manager-role webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
mkdir -p /Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/testbin
test -f /Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/testbin/setup-envtest.sh || curl -sSLo /Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/testbin/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.8.3/hack/setup-envtest.sh
source /Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/testbin/setup-envtest.sh; fetch_envtest_tools /Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/testbin; setup_envtest_env /Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/testbin; go test ./... -coverprofile cover.out
Using cached envtest tools from /Users/ahameed/go/src/github.com/CrunchyData/crunchy-bridge-operator/testbin
setting up env vars
?       github.com/CrunchyData/crunchy-bridge-operator  [no test files]
?       github.com/CrunchyData/crunchy-bridge-operator/apis/crunchybridge/v1alpha1      [no test files]
?       github.com/CrunchyData/crunchy-bridge-operator/apis/dbaas.redhat.com/v1alpha1   [no test files]
ok      github.com/CrunchyData/crunchy-bridge-operator/controllers/crunchybridge        16.895s coverage: 0.0% of statements
ok      github.com/CrunchyData/crunchy-bridge-operator/controllers/dbaas.redhat.com     20.995s coverage: 30.1% of statements
?       github.com/CrunchyData/crunchy-bridge-operator/internal/bridgeapi       [no test files]
?       github.com/CrunchyData/crunchy-bridge-operator/internal/kubeadapter     [no test files]
```

### CODE REVIEW

**Review by tchughesiv — CHANGES_REQUESTED** [review]
needs command added to README after this line - https://github.com/CrunchyData/crunchy-bridge-operator/blob/9da6b0b4668c8d166cd96fd52017786a71fd0365/README.md?plain=1#L82
something like - 
`kubectl label secret crunchy-bridge-api-key type=credentials -n crunchy-bridge-operator-system`

**Comment by redhatHameed** [comment]
> needs command added to README after this line -
> 
> https://github.com/CrunchyData/crunchy-bridge-operator/blob/9da6b0b4668c8d166cd96fd52017786a71fd0365/README.md?plain=1#L82
> 
> 
> something like -
> `kubectl label secret crunchy-bridge-api-key type=credentials -n crunchy-bridge-operator-system`

updated

**Comment by tchughesiv** [comment]
@redhatHameed be sure this is tested within confines of a crunchy-only deployment. in other words, no dbaas operator involvement. make sure we're not missing anything in README.

File: go.mod [hunk 1]
```diff
@@ -3,19 +3,17 @@ module github.com/CrunchyData/crunchy-bridge-operator
...
-	k8s.io/client-go v0.21.1
-	k8s.io/utils v0.0.0-20210709001253-0e1f9d693477
-	sigs.k8s.io/controller-runtime v0.9.0
+	github.com/onsi/gomega v1.15.0
+	k8s.io/api v0.22.1
+	k8s.io/apiextensions-apiserver v0.22.1
+	k8s.io/apimachinery v0.22.1
+	k8s.io/client-go v0.22.1
+	k8s.io/utils v0.0.0-20210802155522-efc7438f0176
+	sigs.k8s.io/controller-runtime v0.10.0
```
Comment (tchughesiv) [inline go.mod:17]: why are you updating the k8s modules? was it required for newcache?

Comment (redhatHameed) [reply go.mod:17]: this is auto-generated

File: apis/dbaas.redhat.com/v1alpha1/zz_generated.deepcopy.go [hunk 2]
```diff
@@ -1,4 +1,3 @@
-//go:build !ignore_autogenerated
```
[the remark this answers is not in the document]
Comment (redhatHameed) [reply apis/dbaas.redhat.com/v1alpha1/zz_generated.deepcopy.go]: agree, I have go1.16 version, regenerated using go1.17.

File: go.mod (hunk 1 above)
Comment (tchughesiv) [reply go.mod:17]: hmm... please revert to former versions and only update dbaas-operator module, unless these changes are required.

Comment (redhatHameed) [reply go.mod:17]: seeing cockroachdb version,  kind of smiliar version
``` require (
	github.com/RHEcosystemAppEng/dbaas-operator v0.1.3
	github.com/go-logr/logr v0.4.0
	github.com/onsi/ginkgo v1.16.4
	github.com/onsi/gomega v1.15.0
	github.com/parnurzeal/gorequest v0.2.16
	k8s.io/api v0.22.1
	k8s.io/apimachinery v0.22.1
	k8s.io/client-go v0.22.1
	k8s.io/utils v0.0.0-20210802155522-efc7438f0176
	moul.io/http2curl v1.0.0 // indirect
	sigs.k8s.io/controller-runtime v0.10.0
)```

is this due to the SKD version? I am using operator-sdk version: "v1.15.0"

Comment (tchughesiv) [reply go.mod:17]: hmm, try 1.10.1

**Comment by redhatHameed** [comment]
> @redhatHameed be sure this is tested within the confines of a crunchy-only deployment. in other words, no DBaaS operator involvement. make sure we're not missing anything in README.

tested and works without DBaaS installation, updated readme add a step for adding lable in secret.

File: go.mod (hunk 1 above)
Comment (redhatHameed) [reply go.mod:17]: same with version SDK 1.10.1. the dependencies generated automatically, keep as it.

Comment (tchughesiv) [reply go.mod:17]: yea, i think that has more to do the the `newcache` code changes... i seem to remember this requiring a k8s module updates

File: main.go [hunk 3]
```diff
@@ -37,6 +40,8 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/healthz"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
+	dbaasv1alpha1 "github.com/RHEcosystemAppEng/dbaas-operator/api/v1alpha1"
```
[the remark this answers is not in the document]
Comment (tchughesiv) [reply main.go:43]: @wilybrace we can remove this and instead opt for local label key/value vars. thanks!

[the remark this answers is not in the document]
Comment (wilybrace) [reply main.go:43]: I'm ok with using local names that match that library in the short term until things can be refactored in the next normal release cycle, but I suspect you'll want to ensure they match.

[the remark this answers is not in the document]
Comment (redhatHameed) [reply main.go]: @wilybrace PR ready for review

**Review by wilybrace — APPROVED** [review]

**Pull request merged**